rholang/src/test/scala/coop/rchain/rholang/interpreter/DeployParamsSpec.scala

```diff
@@ -45,21 +45,19 @@ class DeployParamsSpec extends fixture.FlatSpec with Matchers {
     implicit val emptyEnv = Env[Par]()
     val ackChannel: Par   = GPrivate(ByteString.copyFrom(rand.next()))
     val empty: Par        = GByteArray(ByteString.copyFrom(new Array[Byte](32)))
-
-    val send             = Send(Runtime.FixedChannels.GET_DEPLOY_PARAMS, List(ackChannel))
-    val shortLeashParams = runtime.shortLeashParams
+    val phloRate: Par     = GInt(98765)
+    val timestamp: Par    = GInt(1234567890)
+    val send              = Send(Runtime.FixedChannels.GET_DEPLOY_PARAMS, List(ackChannel))
+    val shortLeashParams  = runtime.shortLeashParams
     val task = for {
-      _ <- shortLeashParams.codeHash.set(empty)
-      _ <- shortLeashParams.phloRate.set(GInt(98765))
-      _ <- shortLeashParams.userId.set(empty)
-      _ <- shortLeashParams.timestamp.set(GInt(1234567890))
+      _ <- shortLeashParams.setParams(empty, phloRate, empty, timestamp)
```

Comment:
can we encapsulate the Ref-s in the params, so that the previous usage is no longer possible?

So we'd have sth like

```scala
trait ShortLeashParams {
  def setParams[F](...): F[Unit]
  def getPhloRate: F[Par]
  def get...
}
```

Reply:
I've had a quick look at the sources - it seems yes we can. Please do so :)

Reply:
Done.